### PR TITLE
[19.03] Fix dns fallback regression

### PR DESCRIPTION
**Pull request by tiborvass**
Fixes https://github.com/moby/moby/issues/41003
fixes https://github.com/moby/moby/issues/41041

Ref https://github.com/moby/libnetwork/pull/2551

Backport of #41008

### CODE REVIEW

**Review by cpuguy83 — APPROVED** [review]
LGTM pending master PR merge.

**Review by tonistiigi — APPROVED** [review]

**Review by AkihiroSuda — APPROVED** [review]

**Comment by thaJeztah** [comment]
Same comment here as on https://github.com/moby/moby/pull/41008#issuecomment-633439844

> @tiborvass I see you reverted the bump (probably to verify the integration-test); could you
>
> - remove the revert
> - swap the "bump" and "test-case", so that `git bisect` doesn't break?
> - move this PR out of draft

File: hack/dockerfile/install/proxy.installer [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # LIBNETWORK_COMMIT is used to build the docker-userland-proxy binary. When
 # updating the binary version, consider updating github.com/docker/libnetwork
 # in vendor.conf accordingly
-: ${LIBNETWORK_COMMIT:=0941c3f409260d5f05cfa6fc68420d8ad45ee483} # bump_19.03 branch
+: ${LIBNETWORK_COMMIT:=71d4d82a5ce50453b1121d95544f0a2ae95bef9b} # bump_18.03 branch
```
Comment (thaJeztah) [inline hack/dockerfile/install/proxy.installer:6]: 😬 `s/18.03/19.03/` ?

**Review by thaJeztah — APPROVED** [review]
LGTM

found one typo in a comment, but not a blocker

**Comment by thaJeztah** [comment]
(moving this out of draft)

**Review by thaJeztah — APPROVED** [review]
LGTM (assuming green)

**Comment by thaJeztah** [comment]
All green now; merging

**Pull request merged**